ember-file-upload/src/system/data-transfer-wrapper.ts

```diff
@@ -43,6 +47,81 @@ export default class DataTransferWrapper {
     return this.files.length ? this.files : this.items;
   }
 
+  async getFilesAndDirectories() {
+    const files: File[] = [];
+
+    const readEntry = async (entry: FileSystemEntry): Promise<File> => {
```

Comment:
This function returns a promise and does not use the `await` keyword.

```suggestion
    const readEntry = (entry: FileSystemEntry): Promise<File> => {
```

Reply:
Good catch. Removed the async keyword. 
There is an eslint rule which can alert these cases, maybe you can add this to you configuration:
https://eslint.org/docs/latest/rules/require-await